File: RobotLib/src/main/java/com/acmerobotics/library/cameraoverlay/message/ReceiveFrameResponse.java
package com.acmerobotics.library.cameraoverlay.message;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;

import java.io.ByteArrayOutputStream;

public class ReceiveFrameResponse extends Response<Bitmap> {
    private Bitmap bitmap;

    public ReceiveFrameResponse() {
        super(MessageType.RECEIVE_FRAME);
    }

    @Override
    public void set(Bitmap value) {
        bitmap = value;
    }

    @Override
    public Bitmap get() {
        return bitmap;
    }

    @Override
    public byte[] payloadToByteArray() {
        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
        bitmap.compress(Bitmap.CompressFormat.JPEG, 50, outputStream);
        return outputStream.toByteArray();
    }

    @Override
    public void payloadFromByteArray(byte[] bytes) {
        bitmap = BitmapFactory.decodeByteArray(bytes, 0, bytes.length);
    }
}

File: CameraOverlay/src/main/java/com/acmerobotics/cameraoverlay/message/ReceiveFrameResponse.kt
package com.acmerobotics.cameraoverlay.message

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import java.io.ByteArrayOutputStream

class ReceiveFrameResponse : Response<Bitmap>(MessageType.RECEIVE_FRAME) {
    private lateinit var bitmap: Bitmap

    override fun payloadFromByteArray(byteArray: ByteArray) {
        bitmap = BitmapFactory.decodeByteArray(byteArray, 0, byteArray.size)
    }

    override fun payloadToByteArray(): ByteArray {
        val outputStream = ByteArrayOutputStream()
        bitmap.compress(Bitmap.CompressFormat.JPEG, 80, outputStream)
        return outputStream.toByteArray()
    }

    override fun set(value: Bitmap) {
        this.bitmap = value
    }

    override fun get(): Bitmap {
        return bitmap
    }
}